splunkloghandler.py

```diff
@@ -0,0 +1,190 @@
+import json
+import logging
+import logging.handlers
+import socket
+import queue
+import threading
+from datetime import datetime, timedelta
+
+StopTimeoutSeconds = 10
+BatchSize = 100
+
+
+class SplunkLogHandler(logging.Handler):
+    """
+    A class which sends records to a Splunk server via its HTTP Event Collector
+    interface
+    """
+
+    def __init__(self, host, url, token=None, secure=False, context=None):
+        logging.Handler.__init__(self)
+        self.host = host
+        self.url = url
+        self.token = token
+        self.secure = secure
+        self.context = context
+
+    def __getSplunkEventDict(self, record):
+        hostname = socket.gethostname()
+
+        splunk_event = {
+            "host": hostname
+        }
+
+        return splunk_event
+
+    def mapLogRecordWithFormat(self, record):
+        splunk_event = self.__getSplunkEventDict(record)
+        splunk_event["event"] = self.format(record)
+        return splunk_event
+
+    def setFormatter(self, fmt):
+        super().setFormatter(fmt)
+
+    def prepare(self, record):
+        data = json.dumps(self.mapLogRecordWithFormat(record))
+        return data
+
+    def send(self, data):
+        if data is None or data == "":
+            return None
+
+        import http.client
+        host = self.host
+        if self.secure:
+            h = http.client.HTTPSConnection(host, context=self.context)
+        else:
+            h = http.client.HTTPConnection(host)
+
+        h.putrequest('POST', self.url)
+        h.putheader("Content-type",
+                    "application/x-www-form-urlencoded")
+        h.putheader("Content-length", str(len(data)))
+        if self.token:
+            s = 'Splunk ' + self.token
+            h.putheader('Authorization', s)
+        h.endheaders()
+
+        h.send(data.encode('utf-8'))
+        return h.getresponse()
+
+    def emit(self, record):
+        """
+        Emit a record.
+
+        Send the record to the Splunk server as a percent-encoded dictionary
+        """
+        try:
+            self.send(self.prepare(record))
+        except Exception:
+            self.handleError(record)
+
+
+class AsyncSplunkLogHandler(SplunkLogHandler):
+    _sentinel = None
```

Comment:
I would like to have another name to sentinel, like stop_sign or something like that.